### PR TITLE
Split web client localization files

### LINKED ISSUE #4251: Split client-side translations into per-locale bundles
The client should only need to download and load the translations for the current and fallback locales instead of everything. 
Even better would be the ability to only load the translations needed.

### PR DESCRIPTION
**Pull request by notbakaneko**
Localization files are now split per-language instead of being all dumped into the same file;
Only the current and fallback locales are loaded now.

closes #4251

### CODE REVIEW

File: extract-locales.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+const path = require('path');
+const mkdirp = require('mkdirp');
+
+const buildPath = path.resolve(__dirname, 'resources/assets/build');
+const localesPath = path.resolve(buildPath, 'locales');
+const messagesPath = path.resolve(buildPath, 'messages.json');
+
+function extractLanguages() {
+  console.log('Extracting localizations...')
+  mkdirp.sync(localesPath);
```
Comment (nanaya) [inline extract-locales.js]: this should be in `writeTranslations` instead. And maybe followed by removing existing `.js` files as it might be filled with unrelated stuff.

File: extract-locales.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  return languages;
+}
+
+function getAllMesssages() {
+  const content = fs.readFileSync(messagesPath);
+
+  return JSON.parse(content);
+}
+
+function runLangJs()
```
Comment (nanaya) [inline extract-locales.js]: I think something like `generateTranslationJson` is more descriptive.

File: extract-locales.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  }
+}
+
+runLangJs();
+writeTranslations(extractLanguages());
+
+// copy lang.js
+fs.copyFileSync(
+  path.resolve(__dirname, 'vendor/mariuzzo/laravel-js-localization/lib/lang.min.js'),
+  path.resolve(__dirname, 'resources/assets/js/lang.js')
```
Comment (nanaya) [inline extract-locales.js]: in `buildPath` instead?

File: .gitignore [hunk 4]
```diff
@@ -82,8 +81,9 @@ sftp-settings.json
 /bootstrap/cache/
 
 /resources/assets/js/messages.js
```
Comment (nanaya) [inline .gitignore]: not needed anymore

File: extract-locales.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+function getAllMesssages() {
+  const content = fs.readFileSync(messagesPath);
+
+  return JSON.parse(content);
+}
+
+function runLangJs()
+{
+  const { spawnSync } = require('child_process');
+  spawnSync('php', ['artisan', 'lang:js', '--json', messagesPath], { stdio: 'inherit' });
```
Comment (nanaya) [inline extract-locales.js]: looks like docker needs updating to handle this as currently `yarn` container doesn't have `php` and vice versa. It's already not quite working smoothly as one needs to be re-run (`yarn` container, to build correct js) after the other (`php` container, to generate the `messages.js`) but at least it works in the end.

I think the separation should be `php-fpm` and `assets-builder` instead of `php` and `yarn`. Then maybe even make language (and route) js generation as part of webpack.

File: extract-locales.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+function getAllMesssages() {
+  const content = fs.readFileSync(messagesPath);
+
+  return JSON.parse(content);
+}
+
+function runLangJs()
+{
+  const { spawnSync } = require('child_process');
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply extract-locales.js]: scoped to function ┐(ツ)┌

File: extract-locales.js (hunk 5 above)
Comment (notbakaneko) [reply extract-locales.js]: Another problem to be solved another time ™️
